src/App.js

```diff
@@ -1,77 +1,39 @@
-// import logo from './logo.svg';
-// import './App.css';
-
-// function App() {
-//   return (
-//     <div className="App">
-//       <header className="App-header">
-//         <img src={logo} className="App-logo" alt="logo" />
-//         <p>
-//           Edit <code>src/App.js</code> and save to reload.
-//         </p>
-//         <a
-//           className="App-link"
-//           href="https://reactjs.org"
-//           target="_blank"
-//           rel="noopener noreferrer"
-//         >
-//           Learn React
-//         </a>
-//       </header>
-//     </div>
-//   );
-// }
-
-// export default App;
-// src/App.js
 import React, { useState } from 'react';
 import { useRef } from 'react';
 import TodoList from './TodoList';
 import { useReducer } from 'react';
+import { useMemo } from 'react';
 import styled from 'styled-components';
 
-function App() {
-  // const [todos, setTodos] = useState([
-  //   { id: 1, text: 'demoText1', completed: false },
-  //   { id: 2, text: 'demoText2', completed: false },
-  // ]);
-
-  const StyledButton=styled.button`
-  background-color: blue;
-  color: white;
-  padding 10px 20px;
-  border: none;
-  border-radius: 5px;
-  cursor: pointer;
-  `;
-
+const StyledButton=styled.button`
+background-color: blue;
+color: white;
+padding 10px 20px;
+border: none;
+border-radius: 5px;
+cursor: pointer;
+`;
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'complete':
+      return state.map((todo)=>todo.id === action.id ? { ...todo, completed: !todo.completed } : todo);
+    case 'update':
+      return [...state, {id:new Date().getTime()+Math.random()*100, text: action.payload, completed: false}];    
+    default:
+      return state;
+  }
+};
 
-  const reducer = (state, action) => {
-    switch (action.type) {
-      case 'complete':
-        const t=state.map((todo)=>todo.id === action.id ? { ...todo, completed: !todo.completed } : todo);
-        return t;
-      case 'update':
-        const u=[...state, {id:new Date().getTime()+Math.random()*100, text: action.payload, completed: false}];
-        return u;
-      // case 'active':
-      //   return state.filter(todo => !todo.completed);
-      // case 'allCompleted':
-      //   return state.filter(todo=>todo.completed);
-      default:
-        return state;
-    }
-  };
-  
-  
-    let [todos, dispatch] = useReducer(reducer, [
+function App() { 
+    const [todos, dispatch] = useReducer(reducer, [
         { id: 1, text: 'demoText1', completed: false },
         { id: 2, text: 'demoText2', completed: false },
       ]);
 
     const [filter, setFilter] = useState('all');
 
-    const filterTodos= todos.filter(todo => {
+    const filterTodos= useMemo(()=> {return todos.filter(todo => {
```

Comment:
For performance reason is better to filter separetly for each filter (all is not needed to be filtered)